In [1]:
%matplotlib widget
from nav_projection import *
from utils import *

scenario = NAVScenarioGeometricGenerator(load_from_file="nav_geometric_state.pkl")
print(show_contract(scenario.l1.steps1234))

Inputs : [duration1,duration2,duration3,duration4,t1_entry,trtd1_entry,ttid1_entry]
Outputs: [output_dv3,output_error1,output_ertd2,output_t1,output_t2,output_t3,output_trtd1,output_trtd2,output_trtd3,output_ttid1,output_ttid2,output_ttid3,t4_exit,trtd4_exit,ttid4_exit]
A: [
  -duration1 <= -0
  -duration2 <= 0
  -duration3 <= 0
  -duration4 <= 0
  -t1_entry <= -0
  -ttid1_entry <= -0
]
G: [
  -duration1 + output_t1 - t1_entry = 0
  -duration2 - output_t1 + output_t2 = 0
  -duration3 - output_t2 + output_t3 = 0
  -duration4 - output_t3 + t4_exit = 0
  output_dv3 <= 5
  -output_dv3 + 0.2 output_ertd2 <= 0
  output_dv3 - 0.3 output_ertd2 <= 0
  0.9 output_dv3 - output_trtd3 + trtd4_exit <= 0
  -1.1 output_dv3 + output_trtd3 - trtd4_exit <= 0
  -output_error1 + output_ertd2 - output_trtd1 = 0
  output_error1 - 0.005 trtd1_entry <= 0
  -output_error1 - 0.005 trtd1_entry <= 0
  -output_trtd1 + output_trtd2 = 0
  output_trtd1 - trtd1_entry = 0
  -output_trtd2 + output_trtd3 = 0
  -output_tti

In [11]:
c = scenario.l1.steps1234.rename_variables([
    ("duration1", "dt1"),("duration2", "dt2"),("duration3", "dt3"),("duration4", "dt4"),
    
    ("trtd1_entry", "trtd_in"), ("ttid1_entry", "ttid_in"), ("t1_entry", "t_in"), 
    
    ("output_t1", "t1"),
    ("output_t2", "t2"),
    ("output_t3", "t3"),
    
    ("output_trtd1", "trtd1_res"),
    ("output_trtd2", "trtd2_res"),
    ("output_trtd3", "trtd3_res"),
    ("output_ttid1", "ttid1_res"),
    ("output_ttid2", "ttid2_res"),
    ("output_ttid3", "ttid3_res"),
    ("output_error1", "err1_res"),
    ("output_ertd2", "ertd2_res"),
    ("output_dv3", "dv3_res"), 
    
    ("t4_exit", "t_out"), 
    ("trtd4_exit", "trtd_out"), 
    ("ttid4_exit", "ttid_out"), 
])
print(show_contract(c))

Inputs : [dt1,dt2,dt3,dt4,t_in,trtd_in,ttid_in]
Outputs: [dv3_res,err1_res,ertd2_res,t1,t2,t3,t_out,trtd1_res,trtd2_res,trtd3_res,trtd_out,ttid1_res,ttid2_res,ttid3_res,ttid_out]
A: [
  -dt1 <= -0
  -dt2 <= 0
  -dt3 <= 0
  -dt4 <= 0
  -t_in <= -0
  -ttid_in <= -0
]
G: [
  -dt1 + t1 - t_in = 0
  -dt2 - t1 + t2 = 0
  -dt3 - t2 + t3 = 0
  -dt4 - t3 + t_out = 0
  dv3_res <= 5
  -dv3_res + 0.2 ertd2_res <= 0
  dv3_res - 0.3 ertd2_res <= 0
  0.9 dv3_res - trtd3_res + trtd_out <= 0
  -1.1 dv3_res + trtd3_res - trtd_out <= 0
  -err1_res + ertd2_res - trtd1_res = 0
  err1_res - 0.005 trtd_in <= 0
  -err1_res - 0.005 trtd_in <= 0
  -trtd1_res + trtd2_res = 0
  trtd1_res - trtd_in = 0
  -trtd2_res + trtd3_res = 0
  -ttid1_res + ttid2_res = 0
  ttid1_res - ttid_in = 0
  -ttid2_res + ttid3_res = 0
  -ttid3_res + ttid_out = 0
]



Inputs : [dt1,dt2,dt3,dt4,t_in,trtd_in,ttid_in]
Outputs: [dv3_res,err1_res,ertd2_res,t1,t2,t3,t_out,trtd1_res,trtd2_res,trtd3_res,trtd_out,ttid1_res,ttid2_res,ttid3_res,ttid_out]
A: [
  dt1 >= 0 
  dt2 >= 0 
  dt3 >= 0 
  dt4 >= 0
  t_in >= 0
  ttid_in >= 0
]
G: [
  t1 = t_in + dt1
  t2 = t1 + dt2
  t3 = t2 + dt3
  t_out = t3 + dt4

  ttid1_res = ttid_in
  ttid2_res = ttid1_res
  ttid3_res = ttid2_res 
  ttid_out = ttid3_res

  trtd1_res = trtd_in
  trtd2_res = trtd1_res
  trtd3_res = trtd2_res 
  0.9 dv3_res <= trtd3_res - trtd_out <= 1.1 dv3_res

  | err1_res | <= 0.005 trtd_in 
  ertd2_res = trtd1_res + err1_res

  dv3_res <= 5
  0.2 ertd2_res <= dv3_res <= 0.3 ertd2_res 
]